app/controllers/purchases_controller.rb

```diff
@@ -0,0 +1,35 @@
+class PurchasesController < ApplicationController
+  def index
+    @category = Category.find(params['category_id'])
+    @purchases = Purchase.where(user_id: current_user.id, category_id: params['category_id'])
+    @total_price = @purchases.sum(:amount)
+  end
+
+  def create
+    @purchase = Purchase.new(purchase_params)
```

Comment:
- [ ] Based on this project's requirements, for a given category, the list of transactions is presented ordered by the most recent. Let's kindly fix this issue.
- [ ] It's also important to add the date of creation of each category and transactions as found in the [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)